pandas/core/shared_docs.py

```diff
@@ -174,6 +174,12 @@
 
 Notes
 -----
+The implementation of groupby is hash-based, meaning in particular that
```

Comment:
Can you move this below the other note? I think the link to the user guide should be the most prominent. It's not a big deal right now, but would be if this section gets expanded more on in the future.